### DISCUSSION TITLE
Problem in kubeapps - I need a Help :)

### DISCUSSION THREAD

**Discussion by Luskan777** [kubeapps/kubeapps#3157] [Q&A] [answered]
I guys

I have a problem with Kubeapps, i settings the kubeapps in cluster kubernetes, and he is returning this error to me:
![Deepin Screenshot_selecionar área_20210715001020](https://user-images.githubusercontent.com/71342479/125792479-3477d6e0-1a06-48f7-a6e8-1e6e2c770cbe.png)

this is the status of the pods in my environment:
![Deepin Screenshot_selecionar área_20210715100412](https://user-images.githubusercontent.com/71342479/125792914-9dab5a5e-6fd8-4230-8c3c-99a3b4179df6.png)

I saw a similar problem in the document, and the solution would be something about the title-proxy, but in my environment there is no title-proxy,.

I liked your project, and I would like to use it, if you can help me I'm very grateful !!!

**Comment by antgamdia** [comment, marked as answer]
Olá!

The 502 error is due, as the logs says, to a pod not starting properly. The nginx fowardning the requests the the different components is unable to redirect the request where it should go.
Why? In this case, it seems it's your postgresql pod which is failing. Since some components require a DB to exist, they will fail if it is not available.

I would suggest re-checking the values.yaml file, just in case there is something wrong that is preventing your database from starting.
As a first step, try just installing Kubeapps with the default values.

**Reply by Luskan777** [reply]
Hi Antonio, 

I already installed Kubeapps with default values ​​following the basic tutorial on the page, I didn't customize values.yaml, I just followed this tutorial

> helm repo add bitnami https://charts.bitnami.com/bitnami
> kubectl create namespace kubeapps
> helm install kubeapps --namespace kubeapps bitnami/kubeapps

the postgresql pod logs with error is this:
![Deepin Screenshot_selecionar área_20210715121358](https://user-images.githubusercontent.com/71342479/125812622-b95b1ea6-e8aa-43bb-a7c5-ade493555ed0.png)

And this is Nginx's
![Deepin Screenshot_selecionar área_20210715121701](https://user-images.githubusercontent.com/71342479/125813126-09f4bfa8-e71e-4a9f-b4a8-c16a2a2d646a.png)

**Reply by antgamdia** [reply]
In that case, the most likely issue is a problem with your cluster resources.
Which cluster are you using? Are you sure you fulfill the minimum requirement for deploying the postgresql chart (disk space, memory, etc)?

**Reply by Luskan777** [reply]
I'm using Cluster Kubernetes version 1.21.2, I'll check if not something with the cluster resources, but I don't think so

**Reply by absoludity** [reply]
Yeah, those logs are showing that (1) the postgres read pod is unable to get a response from `kubeapps-postgresql:5432` (the master), and (2) Nginx is unable to reach the kubernetes API (`kubernetes.default`).

It seems something with your internal cluster networking is not working as expected. I'd recommend taking a look at https://kubernetes.io/docs/tasks/debug-application-cluster/debug-service/#does-the-service-work-by-dns-name to check the internal DNS names.

**Reply by Luskan777** [reply]
Hi Guys,
Sorry for taking so long to respond, like absolutity
spoke, my problem was dns resolution of my cluster, after solving the dns problem kubeapps is working correctly, thank you very much for your help.
And I can't fail to thank and praise this amazing project :)

**Reply by absoludity** [reply]
Excellent, glad you got it sorted.
